src/main/java/uk/gov/pay/ledger/transaction/dao/mapper/TransactionWithParentMapper.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.pay.ledger.transaction.dao.mapper;
+
+import org.jdbi.v3.core.mapper.RowMapper;
+import org.jdbi.v3.core.statement.StatementContext;
+import uk.gov.pay.ledger.transaction.entity.TransactionEntity;
+import uk.gov.pay.ledger.transaction.state.TransactionState;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+
+public class TransactionWithParentMapper implements RowMapper<TransactionEntity> {
+
+    @Override
+    public TransactionEntity map(ResultSet rs, StatementContext ctx) throws SQLException {
+
+        TransactionEntity parentTransactionEntity = new TransactionEntity.Builder()
+                .withId(getLongWithNullCheck(rs, "parent_id"))
+                .withGatewayAccountId(rs.getString("parent_gateway_account_id"))
+                .withExternalId(rs.getString("parent_external_id"))
+                .withParentExternalId(rs.getString("parent_parent_external_id"))
+                .withAmount(getLongWithNullCheck(rs, "parent_amount"))
+                .withReference(rs.getString("parent_reference"))
+                .withDescription(rs.getString("parent_description"))
+                .withState(getState(rs, "parent_state"))
+                .withEmail(rs.getString("parent_email"))
+                .withCardholderName(rs.getString("parent_cardholder_name"))
+                .withCreatedDate(getZonedDateTime(rs, "parent_created_date").orElse(null))
+                .withTransactionDetails(rs.getString("parent_transaction_details"))
+                .withEventCount(rs.getInt("parent_event_count"))
```

Comment:
Shouldn't it be checking for nulls as we do for other values [for example `getLongWithNullCheck`]? (otherwise it will default to `0` afaik)